### PR TITLE
Make nh work with eye

### PR DESCRIPTION
**Pull request by MegaPiggy**
- [x] Add an astro object component to eye and vessel
- [x] Create planets that can orbit eye
- [x] Turn off observatory and forest of galaxies during vessel and eye surface states
- [x] Fix funny bugs

### CODE REVIEW

File: NewHorizons/Builder/Body/GeometryBuilder.cs [hunk 1]
```diff
@@ -12,8 +12,16 @@ public static GameObject Make(GameObject planetGO, Sector sector, float groundSc
             groundGO.transform.parent = sector?.transform ?? planetGO.transform;
             groundGO.transform.localScale = new Vector3(groundScale, groundScale, groundScale);
             groundGO.transform.position = planetGO.transform.position;
-            groundGO.GetComponent<MeshFilter>().mesh = SearchUtilities.Find("CloudsTopLayer_GD").GetComponent<MeshFilter>().mesh;
```
Comment (xen-42) [inline NewHorizons/Builder/Body/GeometryBuilder.cs:15]: Is the top layer not literally just a sphere? This is original marshmallow code so I'm not sure. If it is we could simplify that.

**Review by xen-42 — CHANGES_REQUESTED** [review]
Overall more comments would be nice bc there's a lot going on. Nice work tho.

File: NewHorizons/Builder/Body/GeometryBuilder.cs (hunk 1 above)
Comment (MegaPiggy) [reply NewHorizons/Builder/Body/GeometryBuilder.cs:15]: It is a sphere but the giant's deep top layer is a rounder sphere than the primitive unity sphere.

**Comment by MegaPiggy** [comment]
I'm gonna change this back to a draft for a little bit

**Comment by MegaPiggy** [comment]
😭

**Comment by MegaPiggy** [comment]
NoOoOo</img>

**Review by Bwc9876 — APPROVED** [review]
lgtm (take this with a grain of salt)

**Review by xen-42 — CHANGES_REQUESTED** [review]
When loading into the eye scene from a previous save I get `Eye state object is NULL!`, when I load in I can only jump, no other movement inputs work.

CloakFieldController and MeteorLauncher spam errors in the main solar system (NH examples installed)
image

Raft Builder fails now
image

Slide reel builder fails now
image

On the off chance that other mods are affecting this, I have Cheat and Debug Menu, Unity Explorer, and Run In Background on

**Comment by MegaPiggy** [comment]
`Eye state object is NULL!` is fine. It doesn't affect anything. I should probably find a way to stop the error though.

The other errors I will look into.

**Comment by MegaPiggy** [comment]
I fixed all those errors.

**Comment by MegaPiggy** [comment]
Okay, constant fluid error is gone.

**Review by xen-42 — CHANGES_REQUESTED** [review]
Material of the smoke orb gets cut off depending on which direction you face it from (I already pinged Megapiggy about this, just adding it here so other people see it too and so I dont forget)

**Comment by xen-42** [comment]
> lgtm (take this with a grain of salt)

if you're not going to test a big PR then don't even leave a review please

**Review by JohnCorby — APPROVED** [review]

**Review by xen-42 — APPROVED** [review]
I am now become death

**Pull request merged**